lib/common/gistUtil.py

```diff
@@ -0,0 +1,157 @@
+
+        logger.debug("Upload file to gist with url:[%s], headers:[%s], data:[%s]" % (upload_url, headers, post_data))
+
+        return NetworkUtil.post_request_and_response(upload_url, post_data, headers, [200, 201])
+
+    def update_existing_gist(self, input_gist_id, input_file_path, input_file_desc="", input_content_type=DEFAULT_CTNT_TYPE_JSON):
+        """
+        Update existing gist
+        @param input_file_path:
+        @param input_file_desc:
+        @return:
+        """
+        if os.path.exists(input_file_path):
+            with open(input_file_path, 'rb') as fh:
+                file_name = os.path.basename(input_file_path)
+                if input_content_type == self.DEFAULT_CTNT_TYPE_JSON:
+                    file_data = json.dumps(json.load(fh)).replace('"', '\\"')
+                else:
+                    file_data = fh.read()
+        else:
+            logger.error("The file[%s] you specify for uploading is not exist!" % input_file_path)
+            return None
+
+        update_url = "%s/gists/%s" % (self.DEFAULT_GITHUB_API_URL, input_gist_id)
+
+        headers = {
+            'Authorization': "token %s" % self.auth_token,
+            'Content-Length': str(os.path.getsize(input_file_path)),
+            'Content-Type': input_content_type
+        }
+
+        post_data = """{
+            "description": "%s",
+            "files":
+                {
+                    "%s": {"content": "%s"}
+                 }
+        }""" % (input_file_desc, file_name, file_data)
+
+        logger.debug("Upload file to gist with url:[%s], headers:[%s], data:[%s]" % (update_url, headers, post_data))
+
+        return NetworkUtil.post_request_and_response(update_url, post_data, headers, [200, 201])
+
+    def list_gists(self):
+        query_url = "%s/users/%s/gists" % (self.DEFAULT_GITHUB_API_URL, self.user_name)
+        return NetworkUtil.get_request_and_response(query_url)
+
+    def generate_gist_file_table(self, input_gist_result_list):
+        return_table_dict = {}
+        for gist_obj in input_gist_result_list:
+            filename_dict = gist_obj.get("files", {})
+            for filename in filename_dict:
+                if filename in return_table_dict:
+                    logger.error("Find duplicate file name [%s] in current user [%s] gist list" % (filename, self.user_name))
+                else:
+                    return_table_dict[filename] = {
+                        "id": gist_obj["id"],
+                        "url": gist_obj["url"],
+                        "raw_url": filename_dict[filename]["raw_url"],
+                        "created_at": gist_obj["created_at"],
+                        "updated_at": gist_obj["updated_at"]
+                    }
+        return return_table_dict
+
+    def upload_file(self, input_file_path, input_file_desc="", input_content_type=DEFAULT_CTNT_TYPE_JSON, input_public_flag=True):
```

Comment:
If there is another method to upload file with `filename` and `content` only, it will be better.